fixes required for linux build

pattakosn opened the pull request:
- forgotten redundant ifdef 
- capitalization issues (linux is case sensitive)
- header path fixes
- ~replace nullptr with VK_NULL_HANDLE~
- ~unused variables~
- ~rename local variables that are hiding outer scope variables~
- ~passing redundant function parameters of variables that are already in the namespace~

edit: updated fixes list to match the submitted commits

PanosK92 commented on build_scripts/premake.lua:
Is there a reason for this? It deviates from the norm, which is third_party/name.

pattakosn replied:
the reason was that as it was, `#include <meshoptimizer/meshoptimizer.h> was needed which (AFAIK) is not the upstreamm way to include it and i couldnt use the meshoptimizer package i have on my linux system (installed with my package manager) the other solution is the one i just pushed: mv the header one dir up

PanosK92 replied:
My understanding is that the package wants it this way, if that's the case then do what you think is best, we can live with one library deviating a bit for now.

PanosK92 commented on runtime/Geometry/GeometryProcessing.h:
Is this needed?

pattakosn replied:
removing the dots?
my understanding is that to get from `runtime/Geometry` to `runtime/RHI/RHI_Vertex.h` you need to go up one level.
This matches the behaviour on my system(compile error).

I am really surprised and curious how it compiled on yours (i will test master on a windows machine tomorrow and ocnfirm that)

pattakosn replied:
The pipeline of the other PR is failing because of this https://github.com/PanosK92/SpartanEngine/actions/runs/13402678342/job/37436651747?pr=171

PanosK92 commented:
Is there a reason for that library path change?

pattakosn commented on runtime/Game/Game.cpp:
```diff
@@ -636,8 +636,8 @@ namespace spartan
                             // make the bark collidable
-                            shared_ptr<PhysicsBody> physics_body = bark->AddComponent<PhysicsBody>();
-                            physics_body->SetShapeType(PhysicsShape::Mesh);
+                            shared_ptr<PhysicsBody> physics_body_bark = bark->AddComponent<PhysicsBody>();
```
Could this be that you are mistakenly redeclaring this here and you are actually hiding the outer one?
Could it be that this physics_body is supposed to be the same as in line 576 for example?

PanosK92 replied:
They will still affect the same memory but the double declaration is a mistake, you can remove physics_body_bark if you want.

PanosK92 commented:
I reviewed the PR and have a question. Most of the changes seem to be variable renames, and in some cases, the new names are longer without adding clarity. I don’t see the benefit of these changes, but I might be missing something, what’s your reasoning?

Also, for future PRs, it’s best to keep them focused. If a PR is about naming changes, it should be just that. If it’s about Linux fixes, then it should only include those. If both are necessary, splitting them into two PRs makes review and merging easier.

pattakosn commented:
i did not realize it at first but you are right, i should keep PRs focused on a single issue. i make this PR specific to fixes required for linux builds. the VK_NULL_HANDLE commit will be submitted with the vulkan fixes (when we get there) and any "cleean" commits in a separate PR

pattakosn commented:
Last night, i modified this MR to include only fixes required for the linux build. Specifically:

- an ifdef about FFX which is no longer needed and breaks compilation on linux.
- capitalization in shaders which again breaks compilation on linux
- modified meshoptimizer header path, again fixing the linux build. the problem was that as it was in spartan, the path to compressonator.h did not match how it is distributed upstream. You can see for example how the package is installed on [debian](https://packages.debian.org/sid/amd64/libmeshoptimizer-dev/filelist) (on arch as well). If you prefer that the local copy of the header remains as it was, i would add an ifdef linux.

PanosK92 commented:
Great, now I can follow this PR, and I can merge it.

PanosK92 commented:
A great example, FXAA wouldn't compile, because it want's the header to be included in a specific way:

https://github.com/PanosK92/SpartanEngine/blob/b5a13b7d33db0e0087b1f1f9dbbb295aa77bf43b/data/shaders/fxaa/fxaa3_11.h#L47

Was that needed for Linux?

pattakosn commented:
> A great example, FXAA wouldn't compile, because it wants the header to be included in a specific way:
> 
> https://github.com/PanosK92/SpartanEngine/blob/b5a13b7d33db0e0087b1f1f9dbbb295aa77bf43b/data/shaders/fxaa/fxaa3_11.h#L47
> 
> Was that needed for Linux?

yes, the filename in the repository is with lower "f" but in the include it was with capital "F". Windows do not differentiate lower/capital letters on the filesystem but linux does and this fails to compile. It makes no difference on windows.
